lib/sidebar.rb

```diff
@@ -29,185 +29,193 @@ def pre_render(page)
+
+      url = page[:url]
+      headers = page[:headers]
+
+      grandchildren = all_child_pages.select do |grandchild_path, _|
+        grandchild_path.include? path and grandchild_path != path and \
+          path.split('/').length > level
       end
 
-      subgroup = ''
-      has_subgroups = !all_subgroups.empty?
-      if value[:headers].any? || !subsubgroup_list.empty?
-        if has_subgroups
-          url = value[:url]
-          active = active?(filename, url, true)
-          # puts "#{url}, #{filename}, #{key}" if active
-          item_class = active || (url.split('/').length > level && filename.start_with?(url)) ? 'nav-subgroup active' : 'nav-subgroup'
-          subgroup << "<li class=\"#{item_class}\">"
-          subgroup << "<div class=\"nav-subgroup-heading\"><i class=\"material-icons\">arrow_right</i><a href=\"#{url}\">#{title}</a></div>"
-          subgroup << '<ul class="nav-ul">'
+      sub_group_markup = ''
+      has_child_pages = !all_child_pages.empty?
 
-          if subsubgroup_list.empty?
-            value[:headers].each do |header|
-              subgroup << "<li class=\"nav-leaf\"><a href=\"#{value[:url]}#{header[:hash]}\">#{header[:title]}</a></li>"
+      if (!headers.nil? && headers.any?) || !grandchildren.empty?
+        if has_child_pages
+          active = filename.active?(url, exact: true)
+          item_class = active || (url.split('/').length > level && filename.start_with?(url)) ? 'nav-subgroup active' : 'nav-subgroup'
+          sub_group_markup << "<li class=\"#{item_class}\">"
+          sub_group_markup << '<div class="nav-subgroup-heading">'
+          sub_group_markup << '<i class="material-icons">arrow_right</i>'
+          sub_group_markup << "<a href=\"#{url}\">#{title}</a>"
+          sub_group_markup << '</div>'
+          sub_group_markup << '<ul class="nav-ul">'
+
+          if grandchildren.empty?
+            headers.each do |header|
+              hash = header[:hash]
+              subtitle = header[:title]
+              sub_group_markup << "<li class=\"nav-leaf\"><a href=\"#{url}#{hash}\">#{subtitle}</a></li>"
             end
           else
-            subgroup_leaf_class = active ? 'nav-leaf nav-subgroup-leaf active' : 'nav-leaf nav-subgroup-leaf'
-            subgroup << "<li class=\"#{subgroup_leaf_class}\"><a href=\"#{value[:url]}\">#{title} overview</a></li>"
+            sub_group_leaf_class = active ? 'nav-leaf nav-subgroup-leaf active' : 'nav-leaf nav-subgroup-leaf'
+            sub_group_markup << "<li class=\"#{sub_group_leaf_class}\"><a href=\"#{url}\">#{title} overview</a></li>"
 
-            subsubgroup_list.each do |subsubgroup_key, subsubgroup_value|
-              subgroup << generateSubgroup(filename, subsubgroup_key, subsubgroup_value, subsubgroup_list, 3)
+            grandchildren.each do |grandchild_path, grandchild_page|
+              sub_group_markup << generate_sub_group(filename, grandchild_path, grandchild_page, grandchildren, 3)
```

Comment:
Codacy found an issue: [Line is too long. [114/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)